### PR TITLE
Gracie/comic detail

### PR DESCRIPTION
**Pull request by artsygracie17**

### CODE REVIEW

File: lucien-ios/Dashboard/DashboardCollectionView/DashboardCollectionViewModel.swift [hunk 1]
```diff
@@ -31,6 +31,10 @@ final class DashboardCollectionViewModel {
     func getComicCount() -> Int {
         return dashboardComics.count
     }
+
```
Comment (ghost) [inline lucien-ios/Dashboard/DashboardCollectionView/DashboardCollectionViewModel.swift]: MINOR Trailing Whitespace Violation: Lines should not have trailing whitespace. [rule](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Atrailing_whitespace)

**Comment by IntrepidBuildServer** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DashboardCollectionView.swift#L86](https://github.com/IntrepidPursuits/lucien-ios/blob/05accdcc0e10dbbc403ef98f7b361be61abc76db/lucien-ios/Dashboard/DashboardCollectionView/DashboardCollectionView.swift#L86): Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [rule](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.401 (0.403) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 13.503 (13.505) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lucien-ios/ComicDetailScreen/ComicDetailScreenViewController.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    @IBOutlet private weak var releaseDateLabel: UILabel!
+    @IBOutlet private weak var publisherLabel: UILabel!
+    @IBOutlet private weak var genreLabel: UILabel!
+    @IBOutlet private weak var conditionLabel: UILabel!
+
+    private var viewModel: ComicDetailViewModel
+    let disposeBag = DisposeBag()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
```
Comment (mayasaxena) [inline lucien-ios/ComicDetailScreen/ComicDetailScreenViewController.swift]: 1) remove this func if only calling super 2) the nav bar here doesn't match the designs

File: lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    init(comic: DashboardComicUser) {
+        dashboardComicUser = comic
+        setObservables()
+    }
+
+    func setObservables() {
+        comicTitle.value = dashboardComicUser.dashboardComic.comicTitle
+        storyTitle.value = dashboardComicUser.dashboardComic.storyTitle
+        volume.value = dashboardComicUser.dashboardComic.volume ?? ""
```
Comment (mayasaxena) [inline lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift]: Should you default to an empty string or the placeholder string if the value is nil?

File: lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+//  Copyright © 2017 Intrepid Pursuits. All rights reserved.
+//
+
+import Foundation
+import RxCocoa
+import RxSwift
+
+final class ComicDetailViewModel {
+
+    private var dashboardComicUser: DashboardComicUser
```
Comment (mayasaxena) [inline lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift]: Why is this a `var`? Is it going to change?

File: lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    let comicImage = Variable(UIImage())
+    let comicTitle = Variable("-")
```
Comment (mayasaxena) [inline lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift]: What happens if we decide to change the placeholder string? You'll have to go through and change every string manually. Instead, save it as a constant and reference that here

File: lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            releaseDate.value = DateFormatter().string(from: date)
+        }
+        publisher.value = dashboardComicUser.dashboardComic.publisher ?? ""
+        condition.value = dashboardComicUser.dashboardComic.condition ?? ""
+        comicImage.value = getImage(imageURL: dashboardComicUser.dashboardComic.comicPhotoURL ?? "")
+    }
+
+    func getImage(imageURL: String) -> UIImage {
+        guard let url = URL(string: imageURL) else { return UIImage() }
+        guard let data = try? Data(contentsOf: url) else {
```
Comment (mayasaxena) [inline lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift]: These `guards` can all be combined into one (I've told you this before). Rule of thumb: if you see multiple guards all doing the same thing in the else, you can combine them into one guard

Comment (artsygracie17) [reply lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift]: ah oops, I copy pasted this function from an earlier version of a different branch before you had me change it -- I'll be more careful moving forward

**Comment by IntrepidBuildServer** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DashboardCollectionView.swift#L86](https://github.com/IntrepidPursuits/lucien-ios/blob/db47fb1f9ca5d4cccf3a9926914c181a222c55ea/lucien-ios/Dashboard/DashboardCollectionView/DashboardCollectionView.swift#L86): Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [rule](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.371 (0.374) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 11.614 (11.617) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by IntrepidBuildServer** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DashboardCollectionView.swift#L86](https://github.com/IntrepidPursuits/lucien-ios/blob/1d3a1cf7fabe9b8c6c3f25301aca5b8410ebbf9f/lucien-ios/Dashboard/DashboardCollectionView/DashboardCollectionView.swift#L86): Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [rule](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.380 (0.383) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.249 (6.250) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lucien-ios/Extensions/UILabel+Extensions.swift [hunk 7]
```diff
@@ -18,4 +18,12 @@ extension UILabel {
             attributedText = attributedString
         }
     }
+
+    func setLineHeight(text: String, lineSpacing: Double) {
+        let attributedString = NSMutableAttributedString(string: text)
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.lineSpacing = CGFloat(lineSpacing)
+        attributedString.addAttribute(NSAttributedStringKey.paragraphStyle, value:paragraphStyle, range:NSMakeRange(0, attributedString.length))
```
Comment (IntrepidBuildServer) [inline lucien-ios/Extensions/UILabel+Extensions.swift]: MINOR Legacy Constructor Violation: Swift constructors are preferred over legacy convenience functions. [rule](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Alegacy_constructor)

**Comment by IntrepidBuildServer** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DashboardCollectionView.swift#L86](https://github.com/IntrepidPursuits/lucien-ios/blob/f10b4cf8dd34fac674e687e5937a198854031851/lucien-ios/Dashboard/DashboardCollectionView/DashboardCollectionView.swift#L86): Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [rule](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+    init(comic: DashboardComicUser) {
+        dashboardComicUser = comic
+        setObservables()
+    }
+
+    func setObservables() {
+        comicTitle.value = dashboardComicUser.dashboardComic.comicTitle
+        storyTitle.value = dashboardComicUser.dashboardComic.storyTitle
+        volume.value = dashboardComicUser.dashboardComic.volume ?? "-"
```
Comment (mayasaxena) [inline lucien-ios/ComicDetailScreen/ComicDetailViewModel.swift]: What's the point of saving this as `emptyDefault` if you're not going to use it?

**Review by mayasaxena — CHANGES_REQUESTED** [review]

**Comment by IntrepidBuildServer** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DashboardCollectionView.swift#L86](https://github.com/IntrepidPursuits/lucien-ios/blob/79cc5162cf8898e18d5d4ddd2d0a2e39cffed495/lucien-ios/Dashboard/DashboardCollectionView/DashboardCollectionView.swift#L86): Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [rule](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-28-CCDGZ7UI3WPCNSSSZZSPSCNTHAEIZ6YXKP4XFZMHCLLOR7OZDWEA/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-28-CCDGZ7UI3WPCNSSSZZSPSCNTHAEIZ6YXKP4XFZMHCLLOR7OZDWEA/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_6Y91T0PWAFKP-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.383 (0.385) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.455 (6.457) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lucien-ios/Extensions/UILabel+Extensions.swift (hunk 7 above)
Comment (mayasaxena) [inline lucien-ios/Extensions/UILabel+Extensions.swift]: You need to fix warnings from this

**Comment by ghost** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DashboardCollectionView.swift#L86](https://github.com/IntrepidPursuits/lucien-ios/blob/d7688a91da13144682bf67beaa80152d240d1e58/lucien-ios/Dashboard/DashboardCollectionView/DashboardCollectionView.swift#L86): Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [rule](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by ghost** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-28-CCDGZ7UI3WPCNSSSZZSPSCNTHAEIZ6YXKP4XFZMHCLLOR7OZDWEA/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-28-CCDGZ7UI3WPCNSSSZZSPSCNTHAEIZ6YXKP4XFZMHCLLOR7OZDWEA/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_6Y91T0PWAFKP-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">There was an error collecting metrics for your project. This generally occurs due to test failures. Make sure all your tests are passing, and re-run the build.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.393 (0.394) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 7.659 (7.661) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Review by mayasaxena — APPROVED** [review]

**Pull request merged**